evalml/automl/callbacks.py

```diff
@@ -0,0 +1,47 @@
+from evalml.exceptions import PipelineScoreError
+from evalml.utils.logger import get_logger
+
+logger = get_logger(__file__)
+
+
+def silent_error_callback(exception, automl, **kwargs):
+    """No-op."""
+    pass
+
+
+def raise_error_callback(exception, automl, **kwargs):
+    """Raises the exception thrown by the AutoMLSearch object. Also logs the exception as an error."""
+    logger.error(f'AutoMLSearch raised a fatal exception: {str(exception)}')
+    raise exception
+
+
+def log_and_save_error_callback(exception, automl, **kwargs):
```

Comment:
I think it would be more useful for users if we save the traceback instead of the exception?

Context: https://github.com/alteryx/evalml/issues/1174#issuecomment-718963326

Reply:
Having the traceback would be great for performance testing as well. Maybe we can save both in a tuple?

Reply:
Can't you compute the traceback given the exception? Or is it true that once you leave scope you can't get the full stack trace anymore?

I suppose either way its not expensive to compute the traceback, so I agree we should do so. Perhaps in that case the args to each callback could be: automl object, exception, traceback, kwargs

Reply:
Thanks for the feedback guys! I updated s.t. the arguments are automl object, exception, traceback, kwargs and both the traceback and exception are logged/saved :)